src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -762,7 +772,8 @@ private class FastqDemultiplexer(val sampleInfos: Seq[SampleInfo],
         )
       }
 
-    DemuxResult(sampleInfo=sampleInfo, numMismatches=numMismatches, records=demuxRecords)
+    val passQc = if (!keepOnlyPassing) true else demuxRecords.forall(d => d.readInfo.forall(_.passQc))
+    DemuxResult(sampleInfo=sampleInfo, numMismatches=numMismatches, records=demuxRecords, passQc=passQc)
```

Comment:
@kstromhaug @tfenne part of me wonders if we should check `keepOnlyPassing` here when setting `passQc` on the `DemuxResult`, but instead check it when we filter the `resultIterator`.